src/test/java/io/fabric8/elasticsearch/integrationtest/ProxyUsernameIntegrationTest.java

```diff
@@ -0,0 +1,65 @@
+/**
+ * Copyright (C) 2015 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.fabric8.elasticsearch.integrationtest;
+
+import org.junit.Test;
+
+import io.fabric8.elasticsearch.plugin.KibanaUserReindexFilter;
+
+public class ProxyUsernameIntegrationTest extends ElasticsearchIntegrationTest {
+    
+    @Test
+    public void testClusterAdminUserWithSimpleNameCreateProfileIndex() throws Exception {
+        startES();
+        
+        givenUserIsClusterAdmin("admin");
+        givenUserIsAdminForProjects("logging", "openshift");
+        
+        whenGettingDocument("_cat/indices");
+        assertThatResponseIsSuccessful();
+    }
+    
+    @Test
+    public void testNormalUserWithSimpleNameCreateProfileIndex() throws Exception {
+        startES();
+        
+        givenUserIsNotClusterAdmin("somerandomuser");
+        givenUserIsAdminForProjects("myproject");
+        
+        whenGettingDocument("_cat/indices");
+        assertThatResponseIsForbidden();
+    }
+    
+    @Test
+    public void testUsersWithSimpleNameCreatesProfileIndex() throws Exception {
+        startES();
+
+        String [] users = {"foo@email.com", 
+            "CN=jdoe,OU=DL IT,OU=User Accounts,DC=example,DC=com", 
+            "test/username",
```

Comment:
this used to be "test\\\\username" before and "CN=Lastname\\\\\\\\ , Firstname...." however it seems SG is converting "\\" to "/" internally... not sure if thats something we should test/fix though?

@jcantrill @richm

Reply:
It has to be tested with the backward slashes as that is directly related to issues we have seen in the field.  We modify that value [1] .  It may be mitigated with the latest code where hashes the values to generate the roles, but they should remain the same because the proxy header will have to match the entry in the rolesmapping.yml

[1] https://github.com/fabric8io/openshift-elasticsearch-plugin/blob/master/src/main/java/io/fabric8/elasticsearch/plugin/OpenshiftRequestContextFactory.java#L91

Reply:
Got it, I needed to do some more escaping for my mock response...